File: testjsp/src/main/java/controller/member/signup.java
package controller.member;

import java.io.IOException;
import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import model.dao.MemberDao;
import model.dto.MemberDto;

@WebServlet("/member/signup") // 패키지/파일명
public class signup extends HttpServlet {
	protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
		request.setCharacterEncoding("UTF-8");

		// 1. JSP(HTML) form 입력받은 데이터 요청!!!
		String mtitle = request.getParameter("mtitle"); 
		String mcontent =  request.getParameter("mcontent");
		String mname =  request.getParameter("mwriter");
		String mpassword =  request.getParameter("mpassword");
		String mpasswordconfirm =  request.getParameter("mpasswordconfirm");
		
		// 2. 변수6개 ---> DTO 객체화 
		MemberDto dto = new MemberDto( 
				0, mtitle,
				mcontent, mname,
				mpassword,null , 0);
		// 3. 통신 확인 [ HTML ---> JAVA ]
		System.out.println( dto.toString() );
		// 4. DAO 메소드 호출용 객체 선언 
			// 1. 싱글톤객체가 아닐경우 [ 1. 객체생성  2. 객체명.메소드명() ] 
				// MemberDao dao = new MemberDao(); 
				// boolean result =  dao.signup(dto);
			// 2. 싱글톤객체가 있을경우 [ 클래스명.getInstance().메소드명() ]
		boolean result = MemberDao.getInstance().signup(dto);
		// 5. 결과 제어
		if( result ) {response.sendRedirect("/testjsp/member/success.jsp");} // 이동시키고 싶은 파일명 
		else {System.out.println();}
		
		
	}
	private static final long serialVersionUID = 1L;
       

    public signup() {
        super();
    }
	protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
        
		
	}


}

File: testjsp/src/main/webapp/js/signup.js
//	<button type="button" onclick="delete()">삭제버튼</button>
//</form>
//
//		<table id="boardTable">
//		<tr>
//			<th id="mno">회원번호 </th>
//			<th id="mtitle">제목 </th>
//			<th id="mcontent">내용 </th>
//			<th id="mname">작성자 </th>
//			<th id="mpassword">비밀번호 </th>
//			<th id="mdate">작성일 </th>
//			<th id="mpoint">조회수 </th>
//		</tr>
//	</table>

	let mtitle = document.querySelector("#mtitle");
	let mpassword = document.querySelector("#mpassword");
	let mcontent = document.querySelector("#mcontent");
	let mwriter = document.querySelector("#mwriter");
signup();
function signup(){
	$.ajax({
		url : "http://localhost:8080/testjsp/member/signup.jsp", 
		data : { 
			"mtitle" : mtitle ,
		    "mpassword" : mpassword,
		     "mcontent" : mcontent,
		      "mwriter" : mwriter   
		} , 
		success : function( result ){
			alert("성공했습니다.");
		}
	})
	//print();
}
//게시판 출력
//function print(){
//	
//}



